components/renku_data_services/crc/db.py

```diff
@@ -466,6 +487,62 @@ async def update_resource_class(
                     )
                 return cls_model
 
+    @_only_admins
+    async def get_tolerations(self, api_user: base_models.APIUser, resource_pool_id: int, class_id: int) -> List[str]:
+        """Get all tolerations of a resource class."""
+        async with self.session_maker() as session:
+            res_classes = await self.get_classes(api_user, class_id, resource_pool_id=resource_pool_id)
+            if len(res_classes) == 0:
+                raise errors.MissingResourceError(
+                    message=f"The resource pool with ID {resource_pool_id} or the resource "
+                    f"class with ID {class_id} do not exist, or they are not related."
+                )
+            stmt = select(schemas.TolerationORM).where(schemas.TolerationORM.resource_class_id == class_id)
+            res = await session.execute(stmt)
+            return [i.key for i in res.scalars().all()]
+
+    @_only_admins
+    async def delete_tolerations(self, api_user: base_models.APIUser, resource_pool_id: int, class_id: int):
+        """Delete all tolerations for a specific resource class."""
+        async with self.session_maker() as session, session.begin():
+            res_classes = await self.get_classes(api_user, class_id, resource_pool_id=resource_pool_id)
+            if len(res_classes) == 0:
+                raise errors.MissingResourceError(
+                    message=f"The resource pool with ID {resource_pool_id} or the resource "
+                    f"class with ID {class_id} do not exist, or they are not related."
+                )
```

Comment:
should delete be idempotent?

Reply:
You mean that this error should not be raised at all? 

I think an error should be raised here because the resource class or pool that contains the toleration does exist. So in this case the caller of this method has very wrong assumptions about what they are doing. It is not that they are trying to delete the same toleration again (this will not error). This error is raised when:
- the resource pool does not exist
- the resource class does not exist
- the resource class does not belong to the pool